pkg/fakeservice/group_service_factory.go

```diff
@@ -29,7 +29,7 @@ func NewNsGroupService(ip, username, password, apiVersion string, synchronous bo
 }
 
 func NewNimbleGroupService(serviceOpts ...service.ServiceOptions) (gs *NsGroupService, err error) {
```

Comment:
Seeing it in this context, it looks like service.ServiceOptions should be renamed to service.ServiceOption.

Reply:
I think we had this conversation in the previous pull request. We changed it from singular to plural because it contains all the services options, i.e. this chunk of code
 
```
// GroupServiceOptions :
type GroupServiceOptions struct {
	Host       string
	WaitOnJob  bool
	Username   string
	Password   string
	IsTenant   bool
	ApiVersion string
}
```

Reply:
You are right about the GroupServiceOptions name.  We discussed it here: https://github.com/hpe-storage/nimble-golang-sdk/pull/53#discussion_r673724104.

This comment is about ServiceOptions, which is a type declaration for a function that handles a single option.  I'm suggesting changing this:

`type ServiceOptions func(*GroupServiceOptions)`

to this:

`type ServiceOption func(*GroupServiceOptions)`

It seems strange for the NewNimbleGroupService() function to take a list of option parameters, each of which is plural.

Reply:
ah okay I see! Thanks for the clarification!